src/libp2p_cache.erl

```diff
@@ -74,8 +74,18 @@ init([TID]) ->
     libp2p_swarm_auxiliary_sup:register_cache(TID),
     SwarmName = libp2p_swarm:name(TID),
     DataDir = libp2p_config:base_dir(TID),
-    Opts = [{file, filename:join([DataDir, SwarmName, "cache.dets"])}],
-    {ok, Dets} = dets:open_file(SwarmName, Opts),
+    CacheFilePath = filename:join([DataDir, SwarmName, "cache.dets"]),
+    Opts = [{file, CacheFilePath}],
+    Open = fun() -> dets:open_file(SwarmName, Opts) end,
+    {ok, Dets} =
+        case Open() of
+            {error, {not_a_dets_file, CacheFilePath}} ->
```

Comment:
Would suggest making this `{error, _}`

Reply:
I'd hesitate to do a specific thing for a general case, but I guess this just a cache file, so the worst scenario isn't that bad :man_shrugging: